src/lsm/lsm_cursor.c

```diff
@@ -1458,6 +1458,12 @@ __clsm_put(WT_SESSION_IMPL *session, WT_CURSOR_LSM *clsm,
 	if (position)
 		clsm->current = primary;
 
+	/* Reserve: only update the primary chunk, ignore the record count. */
+	if (reserve) {
+		primary->set_key(primary, key);
+		return (primary->reserve(primary));
+	}
```

Comment:
I think we need to reserve on the same set of chunks that we would update in (i.e., move this into the loop below).  Otherwise, we could miss a conflict with a concurrent transaction and a subsequent `update` can fail unexpectedly.